Mixpanel/MPABTestDesignerConnection.m

```diff
@@ -255,21 +259,78 @@ - (void)webSocket:(MPWebSocket *)webSocket didCloseWithCode:(NSInteger)code reas
+        self.connectivityIndicatorWindow.backgroundColor = [UIColor clearColor];
+        self.connectivityIndicatorWindow.windowLevel = UIWindowLevelAlert;
+        self.connectivityIndicatorWindow.alpha = 0;
+        [self.connectivityIndicatorWindow setHidden:NO];
+        UIImageView *bubbleView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, CONNECTIVITY_INDICATOR_SIZE, CONNECTIVITY_INDICATOR_SIZE)];
+        bubbleView.contentMode = UIViewContentModeScaleAspectFill;
+        bubbleView.image = [MPResources imageNamed:@"MPConnectivityIndicator"];
+        [self.connectivityIndicatorWindow addSubview:bubbleView];
+        
+        UIPanGestureRecognizer *panRecognizer = [[UIPanGestureRecognizer alloc] initWithTarget:self action:@selector(moveConnectivityIndicator:)];
+        [self.connectivityIndicatorWindow addGestureRecognizer:panRecognizer];
+        indicatorAnimator = [[UIDynamicAnimator alloc] initWithReferenceView:self.connectivityIndicatorWindow];
+        
+        [UIView animateWithDuration:0.3 animations:^{
+            self.connectivityIndicatorWindow.alpha = 1;
+        }];
     }
 }
 
 - (void)hideConnectedView
 {
-    if (_recordingView) {
-        [_recordingView removeFromSuperview];
+    if (self.connectivityIndicatorWindow) {
+        [self.connectivityIndicatorWindow setHidden:YES];
+    }
+    self.connectivityIndicatorWindow = nil;
+}
+
+- (void)moveConnectivityIndicator:(UIPanGestureRecognizer *)gesture
+{
+    if (!self.connectivityIndicatorWindow) {
+        return;
+    }
+    
+    CGPoint currPoint = [gesture locationInView:gesture.view];
+    switch (gesture.state) {
+        case UIGestureRecognizerStateBegan: {
+            [indicatorAnimator removeAllBehaviors];
+            indicatorPrevPoint = currPoint;
+            break;
+        }
+        case UIGestureRecognizerStateEnded: {
+            UIWindow *mainWindow = [UIApplication sharedApplication].keyWindow;
+            CGPoint velocity = [gesture velocityInView:mainWindow];
+            CGFloat magnitude = sqrt((velocity.x * velocity.x) + (velocity.y * velocity.y));
+            
+            UICollisionBehavior* collisionBehavior = [[UICollisionBehavior alloc] initWithItems:@[self.connectivityIndicatorWindow]];
+            UIBezierPath *path = [UIBezierPath bezierPathWithRect:[UIScreen mainScreen].bounds];
+            [collisionBehavior addBoundaryWithIdentifier:@"border" forPath:path];
+            [indicatorAnimator addBehavior:collisionBehavior];
+            
+            UIPushBehavior *pushBehavior = [[UIPushBehavior alloc]
+                                            initWithItems:@[self.connectivityIndicatorWindow]
+                                            mode:UIPushBehaviorModeInstantaneous];
+            pushBehavior.pushDirection = CGVectorMake((velocity.x / 10) , (velocity.y / 10));
```

Comment:
Since these are vectors, I don't think we want to divide by `10`, instead we need to normalize the vector to unit length since it is indicating direction.